DOC-3498: tinymceai on-premises documentation (#4142)
tinymce/tinymce-docs: modules/ROOT/pages/tinymceai-on-premises-jwt.adoc

        SecretKey key = Keys.hmacShaKeyFor(apiSecret.getBytes(StandardCharsets.UTF_8));
        Instant now = Instant.now();

        String token = Jwts.builder()
                .audience().add(envId).and()
                .issuedAt(java.util.Date.from(now))
                .expiration(java.util.Date.from(now.plusSeconds(3600)))
                .subject(user.getUsername())
                .claim("user", Map.of(
                        "name",  user.getUsername(),
                        "email", ""
                ))
                .claim("auth", Map.of(
                        "ai", Map.of(
                                "permissions", List.of(
                                        "ai:conversations:*",
                                        "ai:models:agent",
                                        "ai:actions:system:*",
                                        "ai:reviews:system:*"
                                )
                        )
                ))
                .signWith(key, Jwts.SIG.HS256)
                .compact();

        return Map.of("token", token);
    }
}
----
====

> == Editor-side token provider

Reviewer: ```suggestion
== Editor-side configuration
```

Author: Kept as "Editor-side token provider" — it's specifically about the `tinymceai_token_provider` callback, not the full editor config (which lives on the Frameworks page). Renaming to "Editor-side configuration" would overlap with the TinyMCE Integration page scope.

Author: Applied — renamed to "Editor-side configuration". See 74aa678.

|Key source |*API Secret* generated for an access key inside an environment through the Management Panel
|Header format |`Authorization: Bearer <token>`
|===

Pin implementations to `HS256`.

=== The API Secret

The API Secret is generated when creating an access key inside an environment, in the Management Panel under *Environments → <env> → Access keys → New access key*.

* It is shown *once* on the creation screen. Copy it immediately into a secret manager such as Vault, AWS Secrets Manager, Doppler, or a local `.env` file.
* If the secret is lost, rotate: create a new access key, deploy the new secret, then revoke the old key.
* This is *not* the `ENVIRONMENTS_MANAGEMENT_SECRET_KEY` environment variable; that one is used for Management Panel logins, not user-facing AI tokens. Mixing them up produces `invalid-jwt-signature`.

=== API Secret compared with `ENVIRONMENTS_MANAGEMENT_SECRET_KEY`

[cols=",,",options="header",]
|===
|Credential |Purpose |Used by
|*API Secret* |Signs user-facing JWTs presented to the AI runtime endpoints. Created per access key inside an environment. |The application token endpoint. Never appears in any management call.

> |*`ENVIRONMENTS_MANAGEMENT_SECRET_KEY`* |Signs Management Panel logins. Set as an environment variable on the AI service container. |The Management Panel UI.

Reviewer: ```suggestion
|*`ENVIRONMENTS_MANAGEMENT_SECRET_KEY`* |Password to log in to the Management Panel. Set as an environment variable on the AI service container. |The Management Panel UI.
```

Author: Applied — updated to "Password to log in to the Management Panel".

|===

These two credentials are unrelated. Using one in place of the other produces `invalid-jwt-signature`.

== Required claims

Every token MUST contain the following claims.

[cols=",,",options="header",]
|===
|Claim |Type |Description
|`aud` |string |The Environment ID, copied from the Management Panel (for example `viOu8BnjJHb0HGK091p`). Type *must be string*, not array; the verifier rejects array-shaped `aud` (the default in some JWT libraries) with `invalid-jwt-payload`.
|`iat` |number |Issued-at, seconds since epoch (UTC).
|`exp` |number |Expiry, seconds since epoch (UTC). Recommend `iat {plus} 3600` for demos, `iat {plus} 900` for production. The server applies 60 seconds of clock-skew leeway; tokens up to 60 seconds past `exp` still verify.
|`sub` |string |Unique, stable user identifier. Conversation history is isolated per-`sub`; do not reuse one `sub` across users or conversations will leak between them.

> |`auth.ai.permissions` |`string[]` |Array of feature permission strings. See the permissions reference below. Wildcards (`*`) are accepted only in the documented positions; the bare string `"*"` is rejected.

Reviewer: `Wildcards (*)` is rendered as `Wildcards (``)`

Author: Fixed — switched to `+*+` so AsciiDoc renders it as monospace without triggering bold formatting.
